x-pack/plugins/ingest_manager/server/routes/epm/handlers.ts

```diff
@@ -136,13 +140,11 @@ export const installPackageFromRegistryHandler: RequestHandler<
   undefined,
   TypeOf<typeof InstallPackageFromRegistryRequestSchema.body>
 > = async (context, request, response) => {
-  const logger = appContextService.getLogger();
   const savedObjectsClient = context.core.savedObjects.client;
   const callCluster = context.core.elasticsearch.legacy.client.callAsCurrentUser;
   const { pkgkey } = request.params;
   const { pkgName, pkgVersion } = splitPkgKey(pkgkey);
   const installedPkg = await getInstallationObject({ savedObjectsClient, pkgName });
-  const installType = getInstallType({ pkgVersion, installedPkg });
```

Comment:
i need to take a look again, but I think `installType` might need to be called before trying to install the package.

Reply:
The `handleInstallPackageFailure` function I added below takes `installedPkg` as a param which is retrieved before the `installPackage` call and passes it to the `getInstallType` function in the failure case. The `installedPkg` variable will have the initial state of the package before the install occurs. So I think it should operate the same as before because `getInstallType` is just accessing the attributes of the passed in saved object and doing some comparisons. I just moved it into the error handler.